sdk/digitaltwins/Azure.DigitalTwins.Core/src/DigitalTwinsClient.cs

```diff
@@ -662,40 +575,25 @@ async Task<Page<IncomingRelationship>> NextPageFunc(string nextLink, int? pageSi
         /// See the asynchronous version of this method for examples.
         /// </seealso>
         public virtual Pageable<IncomingRelationship> GetIncomingRelationships(string digitalTwinId, CancellationToken cancellationToken = default)
+            => PageableHelpers.CreateEnumerable(GetIncomingRelationshipsImplAsync(false, digitalTwinId, cancellationToken).EnsureSyncEnumerable(),
+                _clientDiagnostics, "DigitalTwinsGeneratedClient.ListIncomingRelationships");
+
+        private async IAsyncEnumerable<Page<IncomingRelationship>> GetIncomingRelationshipsImplAsync(bool async, string digitalTwinId, [EnumeratorCancellation] CancellationToken cancellationToken)
         {
-            Page<IncomingRelationship> FirstPageFunc(int? pageSizeHint)
-            {
-                using DiagnosticScope scope = _clientDiagnostics.CreateScope("DigitalTwinsGeneratedClient.ListIncomingRelationships");
-                scope.Start();
-                try
-                {
-                    Response<IncomingRelationshipCollection> response = _dtRestClient.ListIncomingRelationships(digitalTwinId, cancellationToken);
-                    return Page.FromValues(response.Value.Value, response.Value.NextLink, response.GetRawResponse());
-                }
-                catch (Exception e)
-                {
-                    scope.Failed(e);
-                    throw;
-                }
-            }
+            Response<IncomingRelationshipCollection> response = async
+                ? await _dtRestClient.ListIncomingRelationshipsAsync(digitalTwinId, cancellationToken).ConfigureAwait(false)
+                : _dtRestClient.ListIncomingRelationships(digitalTwinId, cancellationToken);
```

Comment:
This conditional logic can be codegen'ed which would simplify handwritten code even further.